### PR TITLE
Progress metrics refactor and rename

### PR DESCRIPTION
**Pull request by arnongilboa**
**What this PR does / why we need it**:
`clone_progress` and `openstack_populator_progress` and `ovirt_progress` metrics are not in the monitoring package. These metrics are not with correct metrics names, based on the kubevirt and Prometheus metrics naming conventions, they are not documented and are not located under `/pkg/monitoring`. After the code refactoring we should not have Prometheus metrics in other places in the code, other than the `/monitoring/metrics` package, and metrics should be registered using operator-observability package.

In addition, we aligned the progress metric names with the metric naming linter rules: `kubevirt_cdi_clone_progress_total`, `kubevirt_cdi_openstack_populator_progress_total`, `kubevirt_cdi_ovirt_progress_total`.

**Which issue(s) this PR fixes**:
Fixes # https://issues.redhat.com/browse/CNV-36845

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/monitoring/metrics/openstack-populator/populator_metrics.go
Comment (sradco) [reply pkg/monitoring/metrics/openstack-populator/populator_metrics.go:15]: If we are generating the metric it should have `kubevirt_cdi_` prefix

File: pkg/monitoring/metrics/ovirt-populator/populator-metrics.go
Comment (sradco) [inline pkg/monitoring/metrics/ovirt-populator/populator-metrics.go:15]: Same. The metric requires kubevirt_cdi_ prefix

File: pkg/monitoring/metrics/openstack-populator/populator_metrics.go
Comment (assafad) [reply pkg/monitoring/metrics/openstack-populator/populator_metrics.go]: Also missing `_total` suffix, since it's a counter.

File: pkg/monitoring/metrics/ovirt-populator/populator-metrics.go
Comment (assafad) [reply pkg/monitoring/metrics/ovirt-populator/populator-metrics.go:15]: Also missing _total suffix, since it's a counter.

**Comment by assafad** [comment]
@arnongilboa We have a metric name linter, can you please list the new metrics in 
https://github.com/kubevirt/containerized-data-importer/blob/main/tools/prom-metrics-collector/metrics_json_generator.go? You can then use it locally with `make lint-metrics` and see what names you need to fix. 
@machadovilaca @avlitman @sradco FYI, when new metrics groups are added, the developer need to add the new group to the metrics names linter.

**Comment by assafad** [comment]
Unrelated to the PR, we have a small issue in https://github.com/kubevirt/containerized-data-importer/blob/d2f3903e974b53a69e42d810f6507f783851408a/tools/prom-metrics-collector/metrics_json_generator.go#L37
Which uses https://github.com/kubevirt/containerized-data-importer/blob/d2f3903e974b53a69e42d810f6507f783851408a/tools/prom-metrics-collector/metrics_json_generator.go#L11
and this lists only one group of metrics.
We need to use `"github.com/machadovilaca/operator-observability/pkg/operatormetrics"` and then `operatormetrics.ListMetrics()`, which lists all registered metrics. We will open a PR for that.

File: pkg/monitoring/metrics/ovirt-populator/populator-metrics.go
Comment (arnongilboa) [reply pkg/monitoring/metrics/ovirt-populator/populator-metrics.go:15]: @bennyz wdyt?

**Comment by bennyz** [comment]
/lgtm

for forklift populators

File: pkg/monitoring/metrics/operator-controller/metrics.go
Comment (assafad) [inline pkg/monitoring/metrics/operator-controller/metrics.go:20]: @arnongilboa is removing all of those because they all do the same: `return operatormetrics.ListMetrics()`. 
Why did we do it instead of calling it directly?

**Comment by assafad** [comment]
Thanks @arnongilboa! LGTM :)
Over to @machadovilaca, as he also had a few comments.

File: pkg/monitoring/metrics/cdi-controller/metrics.go
Comment (assafad) [inline pkg/monitoring/metrics/cdi-controller/metrics.go:22]: @machadovilaca, @arnongilboa is removing all of those because they all do the same: `return operatormetrics.ListMetrics()`. 
Why did we do it in the first place, instead of calling it directly?

**Review by awels — APPROVED** [review]
Will wait for the metrics folks to give their blessing, but code wise looks good to me.

**Review by machadovilaca — COMMENTED** [review]
nit: I think we could avoid calling `SetupMetrics` in the `init` functions, but let's not block this because of that

File: doc/metrics.md
Comment (sradco) [inline doc/metrics.md:27]: @arnongilboa shouldn't the metric name include populator like it is for openstack?

Comment (sradco) [reply doc/metrics.md:27]: @arnongilboa you update the metrics description to be clearer? The are the same for both the openstack and ovirt metrics

Comment (arnongilboa) [reply doc/metrics.md:27]: @bennyz please consider doing it in a follow-up PR

Comment (sradco) [reply doc/metrics.md:27]: Sure. I'm OK with that.

**Comment by sradco** [comment]
/lgtm

kubevirt_cdi_ovirt_progress_total metric name and description would be updated in a follow-up PR.
This PR is urgent to unblock the metrics code refactoring linter PR.

**Pull request merged**
